src/telemetry_poller_vm.erl

```diff
@@ -0,0 +1,36 @@
+-module(telemetry_poller_vm).
+
+-export([
+  memory/0,
+  total_run_queue_lengths/0
+]).
+
+-spec memory() -> ok.
+memory() ->
+  Measurements = erlang:memory(),
+  telemetry:execute([vm, memory], maps:from_list(Measurements), #{}).
+
+-spec total_run_queue_lengths() -> ok.
+total_run_queue_lengths() ->
+  Total = cpu_stats(total),
+  CPU = cpu_stats(cpu),
+  telemetry:execute([vm, total_run_queue_lengths], #{
+    total => Total,
+    cpu => CPU,
+    io => Total - CPU},
+    #{}).
+
+-ifdef(OTP_RELEASE).
+  -if(?OTP_RELEASE >= 20).
+    -spec cpu_stats(total | cpu) -> non_neg_integer().
+    cpu_stats(total) ->
+      erlang:statistics(total_run_queue_lengths_all);
+    cpu_stats(cpu) ->
+      erlang:statistics(total_run_queue_lengths).
+  -endif.
+-else.
+  % OTP_RELEASE macro is available in OTP 19+ so this covers 18 & 19
```

Comment:
I think this covers only 18. You will need to also add this code as an `else` to the inner `if` above to also cover 19.